chai_lab/chai1.py

```diff
@@ -290,14 +295,29 @@ def run_inference(
 
     # Load structure context
     chains = load_chains_from_raw(fasta_inputs)
+    del fasta_inputs  # Do not reference inputs after creating chains from them
+
     merged_context = AllAtomStructureContext.merge(
         [c.structure_context for c in chains]
     )
     n_actual_tokens = merged_context.num_tokens
     raise_if_too_many_tokens(n_actual_tokens)
 
-    # Load MSAs
-    if msa_directory is not None:
+    # Generated and/or load MSAs
+    if msa_server:
+        assert msa_directory is None, "Cannot specify both MSA server and directory"
+        protein_sequences = [
+            chain.entity_data.sequence
+            for chain in chains
+            if chain.entity_data.entity_type == EntityType.PROTEIN
+        ]
+        msa_dir = output_dir / "mmseqs_msas"
```

Comment:
Yes, it does. Should we use a tmpdir to avoid this?

Reply:
The chai1 code here asserts that `output_dir` should be empty at creation time with `mkdir(exist_ok=False)` and `generate_colabfold_msas` also asserts that the given directory is empty.

I think making a temporary directory at the level of the `generate_colabfold_msas` is tricky because the files have to live outside the scope of that function; I think we can instead make a temporary directory when we call `generate_colabfold_msas`.

Reply:
On the other hand, it might be useful for users to be able to inspect MSAs that came back. It could also be cool to cache the MSAs to avoid regeneration, but we probably don't want to encourage users to clobber prior results.

Maybe the real solution here is to require that the output directory is entirely empty in the first place..

Reply:
I'm happy with temporary directory for now (which is already implemented) - we can explore other options later.

Reply:
Ok - let me implement output directory is empty.

Reply:
Added a check that the output directory should always beempty.